File: C#/Monotonic Stack/901. Online Stock Span.cs
/* 901. Online Stock Span
Difficulty: Medium
Submission Link: https://leetcode.com/problems/online-stock-span/submissions/1384560941/ */

public class StockSpanner {
    Stack<(int, int)> stack = new();

    public StockSpanner() {
        
    }
    
    public int Next(int price) {
        int ans = 1;
        while(stack.Count>0 && stack.Peek().Item1<=price)
        {
            ans+= stack.Pop().Item2;
        }

        stack.Push((price, ans));

        return ans;
    }
}

File: JavaScript/Monotonic Stack/901. Online Stock Span.js
/* 901. Online Stock Span
Difficulty: Medium
Submission Link: https://leetcode.com/problems/online-stock-span/submissions/1384557170/ */

var StockSpanner = function() {
    this.stock = []
};

StockSpanner.prototype.next = function(price) {
     this.stock.push(price)
    for(let i=this.stock.length-2;i>=0;i--){

        if(this.stock[i] > price){
            return this.stock.length-1 - i
        }
    }
    return this.stock.length
};